unstructured/ingest/connector/weaviate.py

```diff
@@ -0,0 +1,160 @@
+
+            return AuthClientPassword(
+                username=username, password=pwd, scope=self.connector_config.auth_keys.get("scope")
+            )
+        return None
+
+    def conform_dict(self, data: dict) -> None:
+        """
+        Updates the element dictionary to conform to the Weaviate schema
+        """
+
+        # Dict as string formatting
+        if record_locator := data.get("metadata", {}).get("data_source", {}).get("record_locator"):
+            # Explicit casting otherwise fails schema type checking
+            data["metadata"]["data_source"]["record_locator"] = str(json.dumps(record_locator))
+
+        # Array of items as string formatting
+        if points := data.get("metadata", {}).get("coordinates", {}).get("points"):
+            data["metadata"]["coordinates"]["points"] = str(json.dumps(points))
+
+        if links := data.get("metadata", {}).get("links", {}):
+            data["metadata"]["links"] = str(json.dumps(links))
+
+        if permissions_data := (
+            data.get("metadata", {}).get("data_source", {}).get("permissions_data")
+        ):
+            data["metadata"]["data_source"]["permissions_data"] = json.dumps(permissions_data)
+
+        # Datetime formatting
+        if date_created := data.get("metadata", {}).get("data_source", {}).get("date_created"):
+            data["metadata"]["data_source"]["date_created"] = date_created + "Z"
+
+        if date_modified := data.get("metadata", {}).get("data_source", {}).get("date_modified"):
+            data["metadata"]["data_source"]["date_modified"] = date_modified + "Z"
+
+        if date_processed := data.get("metadata", {}).get("data_source", {}).get("date_processed"):
+            data["metadata"]["data_source"]["date_processed"] = date_processed + "Z"
+
+        if last_modified := data.get("metadata", {}).get("last_modified", {}):
+            data["metadata"]["last_modified"] = last_modified + "Z"
+
+        # String casting
+        if version := data.get("metadata", {}).get("data_source", {}).get("version"):
+            data["metadata"]["data_source"]["version"] = str(version)
+
+        if page_number := data.get("metadata", {}).get("page_number"):
+            data["metadata"]["page_number"] = str(page_number)
+
+        if regex_metadata := data.get("metadata", {}).get("regex_metadata"):
+            data["metadata"]["regex_metadata"] = str(json.dumps(regex_metadata))
+
+    def write_dict(self, *args, json_list: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+        logger.info(
+            f"writing {len(json_list)} objects to destination "
+            f"class {self.connector_config.class_name} "
+            f"at {self.connector_config.host_url}",
+        )
+        self.client.batch.configure(batch_size=self.write_config.batch_size)
+        with self.client.batch as b:
+            for e in json_list:
+                self.conform_dict(e)
+                b.add_data_object(
+                    {
```

Comment:
can this simply be everything other than the embeddings? In case fields are ever added in the future, making sure those get pushed here as well. Unless this is schema enforced.